src/libasr/diagnostics.cpp

```diff
@@ -78,6 +81,7 @@ std::string Diagnostics::render(const std::string &input,
         } else {
             throw LCompilersException("Error format not supported.");
         }
+        lm.files.pop_back();
```

Comment:
This seems to be using the files from the end of the stack, but I don't think that's a correct assumptions. The locations in a given error message can point to any file.

Reply:
I agree, I faced this situation while passing the wrong arguement for the function